field/src/extension/binomial_extension.rs

```diff
@@ -539,6 +530,22 @@ where
     }
 }
 
+impl<AF, const D: usize> BinomialExtensionField<AF, D>
+where
+    AF: AbstractField,
+    AF::F: BinomiallyExtendable<D>,
+{
+    /// Panics if D2 != D.
+    #[inline]
+    fn from_array_trust_me<const D2: usize>(mut arr: [AF; D2]) -> Self {
+        // One is tempted to do something unsafe here, but it seems like LLVM is
+        // fine getting rid of this.
+        let mut me = Self::default();
+        me.value.swap_with_slice(&mut arr);
+        me
+    }
```

Comment:
I have a mild to strong preference to avoid adding this function (particularly as it's easy to avoid in this case).

Just pass a mutable slice of length D (which will always be 3) to cubic_mul/cubic_square.